scripts/log_ios_binary_metrics.js

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env node
+
+const fs = require('fs');
+const zlib = require('zlib');
+const AWS = require('aws-sdk');
+var s3 = new AWS.S3();
+
+const date = new Date();
+
+// Name of variant and path to binary
+const binaries = [
+  ["universal", "build/ios/pkg/dynamic/Mapbox-stripped"],
+  ["armv7", "build/ios/pkg/dynamic/Mapbox-stripped-armv7"],
+  ["arm64", "build/ios/pkg/dynamic/Mapbox-stripped-arm64"],
+  ["x86_64", "build/ios/pkg/dynamic/Mapbox-stripped-x86_64"]
+]
+
+const iosMetrics = binaries.map(binary => {
+  return JSON.stringify({
+      'sdk': 'maps',
+      'platform' : 'iOS',
+      'arch': binary[0],
+      'size' : fs.statSync(binary[1]).size,
+      'created_at': `${date.getUTCFullYear()}-${date.getUTCMonth() + 1}-${date.getUTCDate()}`
+  })
+}).join('\n');
+
+// Since the CircleCI default workflow runs several jobs for multiple platforms 
+// on each commit, we need to check if binary size metrics for this commit
+// exist already to prevent existing metrics from being overridden.
+
+s3.getObject({
+  Bucket: 'mapbox-loading-dock', 
+  Key: `raw/nadia_staging_test_v2/${process.env['CIRCLE_SHA1']}.json.gz`
+}, (getObjectError, androidData) => {
```

Comment:
Should this say `iosData`?

Reply:
Not at this step, because we're checking if there already is existing Android metrics before uploading new ones, to prevent one from overriding the other. I renamed this variable though to make it clearer.